scipy/stats/_covariance.py

```diff
@@ -0,0 +1,194 @@
+from functools import cached_property
+
+import numpy as np
+from scipy import linalg
+
+
+__all__ = ["CovViaPrecision"]
+
+
+class Covariance:
+    """
+    Representation of a covariance matrix
```

Comment:
This is really missing some documentation IMHO.

Also, from the naming of the subclasses, it seems that classmethods would be more appropriate. Then you would have `Covariance.from_precision` instead of `CovViaPrecision`.

Reply:
It wasn't public before. The subclasses had more information about how they were working. When we go to the classmethod idea, I can add some more information to the `Covariance` class itself.